datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -2251,6 +2251,59 @@ mod tests {
         scan_format(state, &*format, &testdata, file_name, projection, limit).await
     }
 
+    /// Test that 0-byte files don't break while reading
+    #[tokio::test]
+    async fn test_read_empty_parquet() -> Result<()> {
```

Comment:
These validate the two reading modes for parquet, + also the schema inference change. Reverting any of those three will cause one or both of these to fail.